patches/minecraft/net/minecraft/world/ChunkCache.java.patch

```diff
@@ -1,5 +1,14 @@
 --- ../src-base/minecraft/net/minecraft/world/ChunkCache.java
 +++ ../src-work/minecraft/net/minecraft/world/ChunkCache.java
+@@ -61,7 +61,7 @@
+     @Nullable
+     public TileEntity func_175625_s(BlockPos p_175625_1_)
+     {
+-        return this.func_190300_a(p_175625_1_, Chunk.EnumCreateEntityType.IMMEDIATE);
++        return this.func_190300_a(p_175625_1_, Chunk.EnumCreateEntityType.QUEUED); // Forge: don't modify world from other threads
```

Comment:
Creating a new tile entity from `getActualState` and other methods that get `ChunkCache` will most likely break things, even if it is not causing a CME.
I think this is a safe fix.